pages/spacedeck/[[...roomId]].js

```diff
@@ -0,0 +1,309 @@
+                for (const sketch of Object.values(object)) {
+                    // if the element does not have an id key, we can safely assume it is a key of a folder from recursion and skip it.
+                    if (!sketch.id) continue;
+                    // now we can check if a sketch already exists in matrix and if so we can skip this sketch.
+                    if (matrixSketches[sketch.id]) {
+                        // we check if the names of our sketches are still matching on the matrix server and on the sketch server
+                        if (sketch.name !== matrixSketches[sketch.id].name) {
+                            // eslint-disable-next-line no-undef
+                            if (process.env.NODE_ENV === 'development') console.log('changing name for ' + matrixSketches[sketch.id]);
+                            await matrixClient.setRoomName(matrixSketches[sketch.id].id, sketch.name);
+                        }
+                        continue;
+                    }
+                    // then we check if the entry is a folder and if so cycle through the folder and underlying sketches first
+                    if (sketch.type === 'folder') {
+                        await updateStructure(sketch, parent);
+                        continue;
+                    }
+                    // otherwise we create a room for the sketch
+                    const link = getConfig().publicRuntimeConfig.authProviders.spacedeck.baseUrl + '/spaces/' + sketch.id;
+                    await createSketchRoom(link, sketch.name, parent);
+                }
+            };
+            const syncSketches = await spacedeck.syncAllSketches()
+                .catch((error) => {
+                    console.debug(error);
+                    setIsSpacedeckServerDown(true);
+                });
+            syncSketches && await updateStructure(spacedeck.getStructure());
+            setSyncingServerSketches(false);
+        };
+
+        if (!cancelled && serviceSpaceId && serverSketches) {
+            syncServerSketchesWithMatrix();
+        }
+
+        return () => cancelled = true;
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [serviceSpaceId, serverSketches]);
+
+    useEffect(() => {
+        let cancelled = false;
+        const populateSketchesfromServer = async (recursion) => {
+            if (!isEmpty(spacedeck.getStructure())) {
+                setServerSketches(spacedeck.getStructure());
+            } else if (!recursion) {
+                await spacedeck.syncAllSketches();
+                populateSketchesfromServer(true);
+            }
+        };
+        !cancelled && getConfig().publicRuntimeConfig.authProviders.spacedeck.baseUrl && populateSketchesfromServer();
+
+        return () => {
+            cancelled = true;
+        };
+    }, [spacedeck]);
+
+    useEffect(() => {
+        let cancelled = false;
+
+        !cancelled && setContent(matrix.roomContents.get(roomId));
```

Comment:
I _think_ we can stop doing this as well... check the first commit of https://github.com/medienhaus/medienhaus-spaces/pull/45 and see if it works here as well...